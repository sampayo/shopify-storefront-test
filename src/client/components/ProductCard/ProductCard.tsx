/* eslint-disable @next/next/no-img-element */
import { Product } from "@/common/models/product";
import React from "react";

interface IProductProps {
  product: Product;
}
function ProductCard(props: IProductProps) {
  const { product } = props;
  return (
    <div key={product.id} className="group relative">
      <div className="aspect-h-1 aspect-w-1 w-full overflow-hidden rounded-md bg-gray-200 lg:aspect-none group-hover:opacity-75 lg:h-80">
        {product.title && product.featuredImage?.url && (
          <img
            src={product.featuredImage?.url}
            alt={product.title}
            className="h-full w-full object-cover object-center lg:h-full lg:w-full"
          />
        )}
      </div>
      <div className="mt-4 flex justify-between">
        <div>
          <h3 className="text-sm text-gray-200">
            <a href="#">
              <span aria-hidden="true" className="absolute inset-0" />
              {product.title}
            </a>
          </h3>
          <p className="mt-1 text-sm text-gray-500">
            from {product.priceRange?.minVariantPrice?.amount}
          </p>
        </div>
      </div>
    </div>
  );
}

export default ProductCard;
