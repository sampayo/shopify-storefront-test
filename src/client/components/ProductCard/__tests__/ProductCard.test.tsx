import renderer from 'react-test-renderer';
import ProductCard from '../ProductCard';

describe('Components', () => {
  describe('ProductCard', () => {
    test('render product card (snapshot)', () => {
      const tree = renderer
        .create(
          <ProductCard
            product={{
              id: '1243',
              handle: 'abc',
              priceRange: {
                minVariantPrice: { amount: '123', currencyCode: 'GBP' },
                maxVariantPrice: { amount: '124', currencyCode: 'GBP' },
              },
              productType: 'product',
              title: 'do proident sunt do id sint nostrud',
            }}
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
