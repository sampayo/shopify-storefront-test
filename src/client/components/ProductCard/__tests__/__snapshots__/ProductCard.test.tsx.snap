// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components ProductCard render product card (snapshot) 1`] = `
<div
  className="group relative"
>
  <div
    className="aspect-h-1 aspect-w-1 w-full overflow-hidden rounded-md bg-gray-200 lg:aspect-none group-hover:opacity-75 lg:h-80"
  />
  <div
    className="mt-4 flex justify-between"
  >
    <div>
      <h3
        className="text-sm text-gray-200"
      >
        <a
          href="#"
        >
          <span
            aria-hidden="true"
            className="absolute inset-0"
          />
          do proident sunt do id sint nostrud
        </a>
      </h3>
      <p
        className="mt-1 text-sm text-gray-500"
      >
        from 
        123
      </p>
    </div>
  </div>
</div>
`;
